File: wear/src/main/java/ca/joshstagg/slate/Slate.java
package ca.joshstagg.slate;

import android.app.Application;

public class Slate extends Application {

    private static Slate instance;
    private ConfigManager mConfigService;

    public Slate() {
        super();
        instance = this;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mConfigService = new ConfigManager(this.getApplicationContext());
    }

    public ConfigManager getConfigService() {
        return mConfigService;
    }

    public static Slate getInstance() {
        return instance;
    }
}

File: wear/src/main/java/ca/joshstagg/slate/Slate.kt
package ca.joshstagg.slate

import android.app.Application
import ca.joshstagg.slate.config.ConfigManager
import timber.log.Timber


class Slate : Application() {

    companion object {
        lateinit var instance: Slate
    }

    val configService: ConfigManager by lazy {
        ConfigManager(applicationContext)
    }

    init {
        instance = this
    }

    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }
}
